x-pack/plugins/reporting/server/routes/lib/get_document_payload.test.ts

```diff
@@ -0,0 +1,148 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { Readable } from 'stream';
+import { UnwrapPromise } from '@kbn/utility-types';
+import { CSV_JOB_TYPE, PDF_JOB_TYPE } from '../../../common/constants';
+import { ReportApiJSON } from '../../../common/types';
+import { getContentStream, statuses } from '../../lib';
+import { createMockConfigSchema, createMockReportingCore } from '../../test_helpers';
+import { jobsQueryFactory } from './jobs_query';
+import { getDocumentPayloadFactory } from './get_document_payload';
+
+jest.mock('../../lib/content_stream');
+jest.mock('./jobs_query');
+
+describe('getDocumentPayload', () => {
+  let getDocumentPayload: ReturnType<typeof getDocumentPayloadFactory>;
+
+  beforeEach(async () => {
+    const schema = createMockConfigSchema();
+    const core = await createMockReportingCore(schema);
+
+    getDocumentPayload = getDocumentPayloadFactory(core);
+
+    (getContentStream as jest.MockedFunction<typeof getContentStream>).mockResolvedValue(
```

Comment:
nit; I know `typeof getContentStream` is not used that often, but I think this would read a bit easier if we just declared

```ts
type GetContentStream = typeof getContentStream
```
above `describe`. Same comment for other cases (like jobsQueryFactory)

Reply:
That's fair notice. I've corrected casting to the return type, so now, it's only one occurrence. I guess that's okay to keep that since we don't need that more than once.